jupyter_book/book_template/content/guide/03_build.md

```diff
@@ -42,16 +42,52 @@ From here, you have **two options**
 
    Building your book's site locally lets you preview your book locally before you
    push it online. It also gives you a bit more functionality than using
-   GitHub Pages to build your book. However, it also requires you to install Ruby.
+   GitHub Pages to build your book. However, it also requires you to install
+   Docker (a containerization platform) _or_ Ruby (an open source programming language).
    If you'd like to build your site locally then jump to the next section.
 
-## Build the book's site HTML locally (optional)
+## Build the book's site HTML locally
 
-If you'd like to build your book's site HTML locally, you'll need to first
-install the necessary dependencies and then build the HTML. Follow these
-steps to do so.
+Once you've generated the markdown for your notebooks and installed the
+necessary dependencies. You are ready to build your site HTML.
+
+Ensure that your notebooks have been converted to markdown, there should be a
+collection of them in `_build/`.
+
+In order to locally build your site's HTML with Jekyll, you'll need to either install
+Docker or Ruby.
+
+In our experience, we've found that Docker provides an easier installation for most systems.
+We therefore generally recommend that you use the provided Docker image.
+To learn more about Docker using Docker, please see the
+[Docker for scientists guide](https://neurohackweek.github.io/docker-for-scientists/)
+
+### Building your site locally with Docker
+
+First, you'll need to make sure you have Docker installed.
+There are [installation instructions for each operating system](https://hub.docker.com/search/?type=edition&offering=community)
+to guide you through this process.
+
+Once Docker is available on your system, you can build the image locally with:
+
+```bash
+docker build -t jupyter-book /PATH/TO/YOUR/CONTENT
+```
 
-### Install the dependencies to preview your book locally
+You can then access this image with:
+
+```bash
+docker run --rm --security-opt label:disable  \
+   -v /PATH/TO/YOUR/CONTENT:/srv/jekyll \
+   -p 4000:4000 \
+   -it -u 1000:1000 \
+   jupyter-book bundle exec jekyll serve --host 0.0.0.0
```

Comment:
I followed these instructions for a Docker build, but for some reason I am encountering a ruby error when executing the `docker run` line :/ 

See screenshots of my terminal here: https://imgur.com/a/MLDSC9X

Did I miss a step, or misunderstand the instructions?

Reply:
I haven't seen that exact error, but just as a guess -- can you delete your local `Gemfile.lock` before running the command ?

Reply:
Ah ha, yes that worked! I saw the messages above about it, but I assumed because they were running older builds of jupyter-book, that deleting it might not be needed for this branch. 

Thanks a lot @emdupre - this feature is **amazing** 🎉🎉🎉🎉 🚀 🚀 🚀 !